lib/pos_blockchain.go

```diff
@@ -119,12 +120,114 @@ func (bc *Blockchain) processOrphanBlockPoS(block *MsgDeSoBlock) error {
 	// is not malformed. If the block is malformed, we should store it as ValidateFailed.
 	if err := bc.isProperlyFormedBlockPoS(block); err != nil {
 		if _, innerErr := bc.storeValidateFailedBlockInBlockIndex(block); innerErr != nil {
-			return errors.Wrapf(innerErr, "processBlockPoS: Problem adding validate failed block to block index: %v", err)
+			return errors.Wrapf(innerErr, "processOrphanBlockPoS: Problem adding validate failed block to block index: %v", err)
+		}
+		return nil
+	}
+	// Validate QC has 2/3s supermajority.
+	utxoView, err := NewUtxoView(bc.db, bc.params, nil, bc.snapshot, nil)
+	if err != nil {
+		// What do we want to do here? We can't validate the QC without a UtxoView.
+		return errors.Wrap(err, "processOrphanBlockPoS: Problem initializing UtxoView")
+	}
+	currentEpochEntry, err := utxoView.GetCurrentEpochEntry()
+	if err != nil {
+		// What do we want to do here? We can't validate the QC without getting the current epoch entry.
+		return errors.Wrap(err, "processOrphanBlockPoS: Problem getting current epoch entry")
+	}
+	var validatorsByStake []*ValidatorEntry
+	if !currentEpochEntry.isBlockHeightInEpoch(block.Header.Height) {
+		// TODO: figure out which epoch its in.
+		if block.Header.Height < currentEpochEntry.InitialBlockHeight {
+			// We can mark it as ValidateFailed. We'll never accept this block.
+			// TODO: Confirm me.
+			if _, innerErr := bc.storeValidateFailedBlockInBlockIndex(block); innerErr != nil {
+				return errors.Wrap(innerErr, "processOrphanBlockPoS: Problem adding validate failed block to block index")
+			}
+			return nil
+		}
```

Comment:
You can remove this check. It's possible for an orphan block to be in the previous epoch and still be valid. This can happen during epoch transitions, where the current tip is the final block of the previous epoch (which automatically transitions us to the new epoch), but we get a conflicting block at the same height as the current tip.

Reply:
Is it okay to only support the "previous" epoch?

Reply:
Yes, we only need to support the previous and next epochs